components/search_results_header/search_results_header.jsx

```diff
@@ -51,16 +62,38 @@ export default class SearchResultsHeader extends React.Component {
             <div className='sidebar--right__header'>
                 <span className='sidebar--right__title'>{this.props.children}</span>
                 <div className='pull-right'>
-                    <button
-                        type='button'
-                        className='sidebar--right__expand btn-icon'
-                        aria-label='Expand'
-                        onClick={this.props.actions.toggleRhsExpanded}
+                    {
+                        this.props.icons.map(
+                            (icon) => {
+                                const id = generateId();
+                                return (
+                                    <OverlayTrigger
+                                        key={id}
+                                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                                        placement='top'
+                                        overlay={<Tooltip id={id}>{icon.tooltip}</Tooltip>}
+                                    >
+                                        <button
+                                            id={id}
+                                            type='button'
+                                            className='sidebar--right__expand btn-icon'
+                                            onClick={() => icon.action()}
```

Comment:
Should we check here if `icon.action` is not null first?

Reply:
I did mention it in the docs that the field is required, however It is better to have a null check around it. Done 👍